frontend/web/src/app/components/digital-print/digital-print-details-panel/digital-print-details-panel.component.ts

```diff
@@ -1,18 +1,87 @@
-import {Component, Input, OnInit} from '@angular/core';
+import {Component, Input, OnChanges, OnInit} from '@angular/core';
+import {DigitalPart} from '../../../model/digital-part';
 import {DigitalPrint} from '../../../model/digital-print';
+import {DigitalPartService} from '../../../services/digital-part/digital-part.service';
 
+declare let $: any;
 
 @Component({
   selector: 'app-digital-print-details-panel',
   templateUrl: './digital-print-details-panel.component.html',
   styleUrls: ['./digital-print-details-panel.component.scss'],
+  providers: [DigitalPartService],
 })
-export class DigitalPrintDetailsPanelComponent implements OnInit {
+export class DigitalPrintDetailsPanelComponent implements OnInit, OnChanges {
+  private table;
   @Input('digitalPrint') digitalPrint: DigitalPrint = null;
+  private digitalParts: DigitalPart[];
 
-  constructor() { }
+  constructor(private digitalPartService: DigitalPartService) {
+
+    /* get parts */
+    this.digitalPartService.getDigitalParts().subscribe((dParts) => {
+      this.digitalParts = dParts;
+    });
+  }
 
   ngOnInit() {
+    this.table = $('#dPartsTable');
   }
 
+  ngOnChanges(){
+    if (this.digitalParts && this.table) { // make sure we truly have loaded the parts
+      this.populate();
+    }
+  }
+
+  private populate() {
+    const _self = this;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "_self". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1854)